subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -1675,30 +1675,17 @@ static void ticker_op_stop_cb(uint32_t status, void *param)
 
 static inline void disable(uint16_t handle)
 {
-	uint32_t volatile ret_cb;
 	struct ll_conn *conn;
-	void *mark;
 	uint32_t ret;
 
 	conn = ll_conn_get(handle);
 
-	mark = ull_disable_mark(conn);
-	LL_ASSERT(mark == conn);
+	ret = ull_ticker_stop_with_mark(TICKER_ID_CONN_BASE + handle,
+					conn, &conn->lll);
 
-	ret_cb = TICKER_STATUS_BUSY;
-	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
-			  TICKER_ID_CONN_BASE + handle,
-			  ull_ticker_status_give, (void *)&ret_cb);
-	ret = ull_ticker_status_take(ret, &ret_cb);
-	if (!ret) {
-		ret = ull_disable(&conn->lll);
-		LL_ASSERT(!ret);
-	}
+	LL_ASSERT(!ret);
 
 	conn->lll.link_tx_free = NULL;
```

Comment:
This particular change assumes that `conn->lll.link_tx_free = NULL;` can be done after unmarking it. Please let me know if that is not the case.

Reply:
yes. no harm here.